internal/actions/action.go

```diff
@@ -0,0 +1,40 @@
+package actions
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type ActionDefinition struct {
+	Type string          `json:"type"`
+	Args json.RawMessage `json:"args"`
+}
+
+type Action interface {
+	Execute() error
+}
+
+// Package-level backend functions that will be set by the caller (due to import cycles)
+var (
+	FetchBuildStatus      func(buildId string) (string, error)
```

Comment:
This is a weird way to avoid import cycles. Check what we do in test-runner and do the same here? Why do this different method here at all?